javav2/example_code/cloudformation/Readme.md

```diff
@@ -0,0 +1,84 @@
+# AWS CloudFormation Java code examples
+
+This README discusses how to run and test the Java code examples for AWS CloudFormation.
+
+## Running the AWS CloudFormation Java files
+
+**IMPORTANT**
+
+The Java examples perform AWS operations for the account and AWS Region for which you've specified credentials, and you may incur AWS service charges by running them. See the [AWS Pricing page](https://aws.amazon.com/pricing/) for details about the charges you can expect for a given service and operation.
+
+Some of these examples perform *destructive* operations on AWS resources, such as deleting a stack. **Be very careful** when running an operation that deletes or modifies AWS resources in your account. It's best to create separate test-only resources when experimenting with these examples.
+
+To run these examples, you'll need the AWS SDK for Java libraries in your **CLASSPATH**.
+
+	export CLASSPATH=target/sdk-examples-1.0.jar:/path/to/aws-java-sdk/<jar-file-name>.jar
+
+Here **/path/to/aws-java-sdk/<jar-file-name>.jar** is the path to where you extracted or built the AWS SDK for Java JAR file.
+
+For systems with Bash support, once you set the **CLASSPATH**, you can run a particular example as follows.
+
+	java com.example.cloudformation.DescribeStacks
+
+
+ ## Testing the AWS CloudFormation files
+
+You can test the Java code examples for AWS CloudFormation by running a test file named **CloudFormationTest**. This file uses JUnit 5 to run the JUnit tests and is located in the **src/test/java** folder. For more information, see [https://junit.org/junit5/](https://junit.org/junit5/).
+
+You can execute the JUnit tests from a Java IDE, such as IntelliJ, or from the command line by using Maven. As each test is ran, you can view messages that inform you if the various tests succeed or fail. For example, the following message informs you that Test 3 passed.
+
+	Test 3 passed
+
+**WARNING**: _Running these JUnit tests manipulates real Amazon resources and may incur charges on your account._
+
+ ### Properties file
+Before running the AWS CloudFormation JUnit tests, you must define values in the **config.properties** file located in the **resources** folder. This file contains values that are required to execute the JUnit tests. For example, you define a stack name. If you do not define all values, the JUnit tests fail.
+
+Define these values to successfully run the JUnit tests:
+
+- **stackName** -  The name of the stack.  
+- **roleARN** – The ARN of the role that has CloudFormation permissions.
+- **location** – The location of file containing the template body. (ie, https://s3.amazonaws.com/<bucketname>/template.yml).
```

Comment:
The location of **the** file containing the template body (**i.e.,** https://s3.amazonaws.com/<bucketname>/teamplate.yml).

Reply:
fixed